clearnode/rpc_router_private.go

```diff
@@ -199,13 +204,15 @@ func (r *RPCRouter) HandleTransfer(c *RPCContext) {
 	}
 
 	// Allow only ledger accounts as destination at the current stage. In the future we'll unlock application accounts.
-	if params.Destination == "" || params.Destination == c.UserID || !common.IsHexAddress(params.Destination) {
+	switch {
+	case params.Destination == "" && params.DestinationTag == "":
+		c.Fail("either destination or destination_tag must be provided")
+		return
+	case params.Destination != "" && !common.IsHexAddress(params.Destination):
```

Comment:
The following should suffice I think?

```suggestion
	case !common.IsHexAddress(params.Destination):
```